enhancements/machine-config/on-cluster-builds.md

```diff
@@ -0,0 +1,1009 @@
+| `machineconfiguration.openshift.io/desiredImage` | The fully-qualified image pullspec for the desired OS image for a given node.          |
+
+#### MachineConfigPools
+
+The MachineConfigPool CRD will be extended thusly:
+
+| **Field Name / Path (JSONPath):**                 | **Optional or Required:**        | **Intended Use:**                                                                                                                                                                                                                                                                                          |
+| ------------------------------------------------- | -------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `.spec.customContent.dockerfile`                  | Optional                         | The Dockerfile intended for use by the BuildController in order to build a customized OS image.                                                                                                                                                                                                            |
+| `.spec.customContent.imageRegistryPushSecretName` | Required if `dockerfile` is set. | Reference to a Kubernetes secret that contains the required registry credentials with sufficient permission to push the image.                                                                                                                                                                             |
+| `.spec.customContent.imageRegistryPullSecretName` | Required if `dockerfile` is set. | Reference to a Kubernetes secret that contains the required registry credentials with sufficient permission to pull both the base image as well as the final OS image. When absent, the global pull secret will be used.                                                                                   |
+| `.spec.customContent.imagePushTarget`             | Required if `dockerfile` is set. | Where to push the resulting image build, e.g. `registry.host.com/org/repo:tag`. The tag will be ignored.                                                                                                                                                                                                   |
+| `.spec.customContent.deployMode`                  | Optional                         | Controls how the MachineConfigs should be combined with the custom OS contents. Possible options are: `image` and `bootc`. See Implementation Details for details.                                                                                                                                         |
```

Comment:
Probably should be hidden from the users as these are implementation details that we handle.

Reply:
I need to rewrite this whole section now that @cdoern and I have thought about what the API could look like.

Reply:
I'm unsure if we want to expose the deployment mode details to the cluster admin though. I think if we decide to offer both `image` and `bootc` deployment modes, a cluster admin would want to make that choice.